roles/translator/src/lib/mod.rs

```diff
@@ -1,7 +1,280 @@
+use async_channel::{bounded, unbounded};
+use futures::FutureExt;
+use rand::Rng;
+pub use roles_logic_sv2::utils::Mutex;
+use status::Status;
+use std::{
+    net::{IpAddr, SocketAddr},
+    str::FromStr,
+    sync::Arc,
+};
+
+use tokio::{
+    sync::broadcast,
+    task::{self, AbortHandle},
+};
+use tracing::{debug, error, info, warn};
+pub use v1::server_to_client;
+
+use proxy_config::ProxyConfig;
+
+use crate::status::State;
+
 pub mod downstream_sv1;
 pub mod error;
 pub mod proxy;
 pub mod proxy_config;
 pub mod status;
 pub mod upstream_sv2;
 pub mod utils;
+
+#[derive(Clone, Debug)]
+pub struct TranslatorSv2 {
+    config: ProxyConfig,
+    reconnect_wait_time: u64,
+}
+
+impl TranslatorSv2 {
+    pub fn new(config: ProxyConfig) -> Self {
+        let mut rng = rand::thread_rng();
+        let wait_time = rng.gen_range(0..=3000);
+        Self {
+            config,
+            reconnect_wait_time: wait_time,
+        }
+    }
+
+    pub async fn start(self) {
+        let (tx_status, rx_status) = unbounded();
+
+        let target = Arc::new(Mutex::new(vec![0; 32]));
+
+        // Sender/Receiver to send SV1 `mining.notify` message from the `Bridge` to the `Downstream`
+        let (tx_sv1_notify, _rx_sv1_notify): (
+            broadcast::Sender<server_to_client::Notify>,
+            broadcast::Receiver<server_to_client::Notify>,
+        ) = broadcast::channel(10);
+
+        let task_collector: Arc<Mutex<Vec<(AbortHandle, String)>>> =
+            Arc::new(Mutex::new(Vec::new()));
+
+        self.internal_start(
+            tx_sv1_notify.clone(),
+            target.clone(),
+            tx_status.clone(),
+            task_collector.clone(),
+        )
+        .await;
+
+        debug!("Starting up signal listener");
+        let task_collector_ = task_collector.clone();
+
```

Comment:
Why did you remove `let mut interrupt_signal_future = Box::pin(tokio::signal::ctrl_c().fuse());` and its management?
I'm just comparing it to what we currently have on `dev` at https://github.com/stratum-mining/stratum/blob/0f0ee1e32a7cdeb9af83718364e0896b36744824/roles/translator/src/main.rs#L83

Reply:
@jbesraa do you have an answer to this? After that, for me it's ready to go

Reply:
sorry, missed this.
fixed